p2p/swarm.go

```diff
@@ -104,34 +106,59 @@ func (s *swarm) waitForGossip() error {
 
 // newSwarm creates a new P2P instance, configured by config, if newNode is true it will create a new node identity
 // and not load from disk. it creates a new `net`, connection pool and discovery.
-func newSwarm(ctx context.Context, config config.Config, newNode bool, persist bool) (*swarm, error) {
+func newSwarm(ctx context.Context, config config.Config, logger log.Log, datadir string) (*swarm, error) {
+
+	if logger == log.NilLogger {
```

Comment:
why have this check? why not decide what kind of logger this func uses by sending it to the func